### PR TITLE
Arachne 1400

### PR DESCRIPTION
**Pull request by mrAlexandr**

### CODE REVIEW

File: src/modules/StudyManager/components/ViewEdit/ModalAddDataSource/VirtualSource/container.js [hunk 1]
```diff
@@ -70,14 +79,18 @@ export default class AddVirtualSourceBuilder {
 
   mapStateToProps(state) {
     const studyData = get(state, 'studyManager.study.data.result');
+    const dataSourceData = get(state, 'studyManager.study.dataSource.data');
```
Comment (johnSamilin) [inline src/modules/StudyManager/components/ViewEdit/ModalAddDataSource/VirtualSource/container.js]: You won't need this var anymore

File: src/modules/StudyManager/components/ViewEdit/ModalAddDataSource/VirtualSource/selectors.js [hunk 2]
```diff
@@ -64,6 +64,28 @@ export default class selectorsBuilder {
...
+    return get(state, 'studyManager.study.dataSource.data.dataNode.dataOwners')
+  }
+
+  getDataSourceOwnerList(currentUserId, dataSourceOwners) {
+    return dataSourceOwners
+      ? dataSourceOwners
+      .map(dataOwner => ({
+        id: dataOwner.id,
+        isRemovable: currentUserId !== dataOwner.id,
+        fullName: dataOwner.firstname + " " + dataOwner.lastname,
```
Comment (johnSamilin) [inline src/modules/StudyManager/components/ViewEdit/ModalAddDataSource/VirtualSource/selectors.js]: Use attributed string

File: src/services/Utils.js [hunk 3]
```diff
@@ -23,15 +23,18 @@
 import errors from 'const/errors';
 import {
   get as _get,
-  isEqual,
+  isEqual
 } from 'lodash';
-import { types as fieldTypes } from 'const/modelAttributes';
+import types, { types as fieldTypes } from 'const/modelAttributes';
```
Comment (johnSamilin) [inline src/services/Utils.js]: duplication

File: src/modules/StudyManager/components/ViewEdit/ListDataSources/presenter.jsx [hunk 4]
```diff
@@ -99,7 +100,10 @@ function DataSourceItem(props) {
           <div {...classes('comment-icon')}>chat_bubble</div>
         </Link>
       }
-      <Link to={dataSource.link}>{dataSource.name}</Link>
+      {dataSource.isVirtual
+        ? <Link onClick={() => editDataSource(dataSource.id)}>{dataSource.name}</Link>
```
Comment (pavgra) [inline src/modules/StudyManager/components/ViewEdit/ListDataSources/presenter.jsx]: add pencil ico inside

File: src/modules/StudyManager/components/ViewEdit/ModalAddDataSource/container.js [hunk 5]
```diff
@@ -34,9 +35,16 @@ export default class ModalAddDataSourceBuilder {
     return ModalAddDataSource;
   }
 
+  mapStateToProps(state) {
+    return {
+      dataSourceId: get(state, 'modal.addDataSource.data.dataSourceId'),
+    }
+  }
+
   build() {
```
Comment (pavgra) [inline src/modules/StudyManager/components/ViewEdit/ModalAddDataSource/container.js]: Use ContainerBuilder

File: src/modules/StudyManager/components/ViewEdit/ModalAddDataSource/VirtualSource/container.js [hunk 6]
```diff
@@ -47,6 +52,13 @@ export class AddVirtualSource extends Component {
     };
   }
 
+  componentWillMount() {
+    const dataSourceId = this.props.dataSourceId;
+    if (dataSourceId) {
+      this.props.loadDataSource({studyId: this.props.studyId, dataSourceId: dataSourceId});
```
Comment (pavgra) [inline src/modules/StudyManager/components/ViewEdit/ModalAddDataSource/VirtualSource/container.js]: { studyId: this.props.studyId, dataSourceId }

File: src/modules/StudyManager/components/ViewEdit/ModalAddDataSource/container.js (hunk 5 above)
Comment (marcos-groupon) [reply src/modules/StudyManager/components/ViewEdit/ModalAddDataSource/container.js]: Disagree: this test already passes each glob path as a separate argv element through ForgeTestCase.mf → sh(), which runs subprocess.run on a list (or a POSIX-quoted string on Windows) with shell=False and sets MSYS=noglob so bash.exe cannot expand * ? [ before the script's set -f. Switching to shlex.quote on a single command string would weaken that argv contract, not strengthen it.

**Pull request merged**